include/aspect/utilities.h

```diff
@@ -1064,6 +1065,41 @@ namespace aspect
          */
         const std::function<Tensor<1,dim> (const Point<dim> &)> function_object;
     };
+
+    /**
+     * Create a permutation vector which can be used by the apply_permutation function
+     * which shorts the vector in increasing order.
+     *
+     * @param vector vector to sort
+     */
+    template <typename T>
+    std::vector<std::size_t>
+    sort_permutation(const std::vector<T> &vector);
+
+    /**
+     * Applies a permutation vector to return a sorted verions of the vector.
```

Comment:
```suggestion
     * Applies a permutation vector to return a sorted version of the vector.
```
?
I also don't understand what this function does: By the name it has nothing to do with sorting but it permutes the given data?

Reply:
Fixed the description